pydantic/type_adapter.py

```diff
@@ -193,7 +116,9 @@ def __init__(
         Args:
             type: The type associated with the `TypeAdapter`.
             config: Configuration for the `TypeAdapter`, should be a dictionary conforming to [`ConfigDict`][pydantic.config.ConfigDict].
-            _parent_depth: depth at which to search the parent namespace to construct the local namespace.
+            _parent_depth: Depth at which to search the for the parent namespace used for schema building.
+                We also use this as a reference level to find the global namespace (_parent_depth - 1).
+                Defaults to 2 because we expect to reference the frame that called the `TypeAdapter` constructor.
```

Comment:
I think it's worth considering - should we default to `1`, then add `1` when we call into `parent_frame_namespace`? That seems more intuitive to me from a user perspective. cc @Viicos.

Reply:
Alrighty, let's leave as is :(